### PR TITLE
GitHub activityrework

### LINKED ISSUE #15: As an author, I want to pull in my github activity to my “stream”

### PR DESCRIPTION
**Pull request by GillisGill**
Has graph and log of github contributions

### CODE REVIEW

File: frontend/src/components/profile/PageLayout.jsx [hunk 1]
```diff
@@ -10,15 +10,34 @@ import EditButton from './EditButton';
 class PageLayout extends React.Component {
     constructor(props) {
         super(props);
-        this.state = { key: 'aboutMe' }
+        this.state = { key: 'aboutMe', items: [], isLoaded:false }
```
Comment (PENGYUXIONG) [inline frontend/src/components/profile/PageLayout.jsx]: This is not mapped from reducer and thus will only render once if in constructor. 

Are you sure it is working properly?

**Review by PENGYUXIONG — COMMENTED** [review]
rest seems okay, just the variable usage is not mapped from reducers. Plz make sure it is working correctly and all the states are changing as you desired.

**Comment by KevinHQi** [comment]
I don't know whether I am using it correctly, but it won't show my page. There seem to be issues with pictures.
image

File: frontend/src/components/profile/PageLayout.jsx [hunk 2]
```diff
@@ -56,7 +79,53 @@ class PageLayout extends React.Component {
                             <AboutMe />
                             <EditButton />
                         </>:
-                    <p>Under developing ...</p>}
+                    <p>
+                        <a href={this.props.profile.github}>
```
Comment (AnkushSharma2698) [inline frontend/src/components/profile/PageLayout.jsx]: All of this github code should be wrapped in like a github component

Comment (AnkushSharma2698) [reply frontend/src/components/profile/PageLayout.jsx]: so as to decouple it from the profile pagelayout

File: frontend/src/components/profile/PageLayout.jsx [hunk 3]
```diff
@@ -10,15 +10,34 @@ import EditButton from './EditButton';
...
 
     handleClick = e => {
         this.setState({key: e.key});
     };    
 
+    componentDidMount() {
+
+        var github_user = (this.props.profile.github.replace("http://github.com/", ''))
+
+        fetch('https://api.github.com/users/' + github_user + '/events')
```
Comment (AnkushSharma2698) [inline frontend/src/components/profile/PageLayout.jsx]: this code requires the username so you need to do generic parsing to get the username from a the url

Comment (AnkushSharma2698) [reply frontend/src/components/profile/PageLayout.jsx]: `PageLayout.jsx:24 GET https://api.github.com/users/https://github.com/AnkushSharma2698/events` This request is a 404 from github and it returns json that is not a list so it is currently an error

Comment (AnkushSharma2698) [reply frontend/src/components/profile/PageLayout.jsx]: 3 things need to be done to genericize the view. 

1. url parsing to get username
2. error handling on a request that does not succeed
3. Move the code to a diff component

**Comment by AnkushSharma2698** [comment]
> rest seems okay, just the variable usage is not mapped from reducers. Plz make sure it is working correctly and all the states are changing as you desired.

not changing state, only reading state and error handliong so should be ok

**Pull request merged**